### PR TITLE
Use SoftwareKeyboardControllerCompat

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**

### CODE REVIEW

**Comment by westnordost** [comment]
Can you explain how this is an improvement? Overall, it is more code. Anything that `SoftwareKeyboardControllerCompat` does better than the current solution?

**Comment by Isira-Seneviratne** [comment]
> Can you explain how this is an improvement? Overall, it is more code. Anything that `SoftwareKeyboardControllerCompat` does better than the current solution?

It calls some new APIs for opening the keyboard introduced in API level 30.

**Comment by westnordost** [comment]
But how is that new API better?

**Comment by Isira-Seneviratne** [comment]
The software keyboard controller API handles some edge cases for showing the keyboard: https://github.com/androidx/androidx/blob/155b4777a51e77095ca1609b1c514d37933d8828/core/core/src/main/java/androidx/core/view/SoftwareKeyboardControllerCompat.java#L97

**Comment by westnordost** [comment]
Nice, that's good to hear.

File: app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt [hunk 1]
```diff
@@ -51,16 +50,13 @@ fun Context.toast(@StringRes resId: Int, duration: Int = Toast.LENGTH_SHORT) {
 fun Context.hasPermission(permission: String): Boolean =
     ContextCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED
 
-fun View.showKeyboard(): Boolean? =
-    context?.inputMethodManager?.showSoftInput(this, SHOW_IMPLICIT)
+fun View.showKeyboard() = SoftwareKeyboardControllerCompat(this).show()
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt:53]: Would you specify the return type for this function and hideKeyboard explicitly?

File: app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/CreateNoteFragment.kt [hunk 2]
```diff
@@ -140,7 +142,11 @@ class CreateNoteFragment : AbstractCreateNoteFragment() {
     override fun onComposedNote(text: String, imagePaths: List<String>) {
         /* pressing once on "OK" should first only close the keyboard, so that the user can review
            the position of the note he placed */
-        if (contentBinding.noteInput.hideKeyboard() == true) return
+        val rootInsets = ViewCompat.getRootWindowInsets(contentBinding.noteInput)
+        if (rootInsets?.isVisible(WindowInsetsCompat.Type.ime()) == true) {
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/CreateNoteFragment.kt:146]: I think this should be a extension function, too then. `fun View.isKeyboardVisible(): Boolean`

**Review by westnordost — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt (hunk 1 above)
Comment (Isira-Seneviratne) [reply app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt:53]: It doesn't return anything.

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/util/ktx/Context.kt:53]: Yes, so it should be `: Unit`

If not specified, the return type is set automatically to whatever the return type of the expression on the right is. For a reader without an IDE that can help him, it is then not obvious what the return type is.

File: app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/CreateNoteFragment.kt [hunk 3]
```diff
@@ -14,6 +14,8 @@ import android.view.animation.BounceInterpolator
 import android.view.animation.TranslateAnimation
 import androidx.core.graphics.toPointF
 import androidx.core.os.bundleOf
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/CreateNoteFragment.kt:18]: Unused imports, right?

```suggestion
```

**Review by westnordost — APPROVED** [review]
Nice, looks good!

**Pull request merged**